ocf_data_sampler/torch_datasets/sample/uk_regional.py

```diff
@@ -22,22 +22,186 @@ def __init__(self, data: NumpySample) -> None:
         return self._data
 
     @override
     def save(self, path: str) -> None:
+        """Saves sample to the specified path in pickle format."""
         # Saves to pickle format
         torch.save(self._data, path)
 
     @classmethod
     @override
     def load(cls, path: str) -> "UKRegionalSample":
+        """Loads sample from the specified path.
+
+        Args:
+            path: Path to the saved sample file.
+
+        Returns:
+            A UKRegionalSample instance with the loaded data.
+        """
         # Loads from .pt format
         # TODO: We should move away from using torch.load(..., weights_only=False)
         return cls(torch.load(path, weights_only=False))
 
+    def _check_dimensions(
+        self,
+        actual_shape: tuple[int, ...],
+        expected_shape: tuple[int, ...],
+        name: str,
+    ) -> None:
+        """Check if dimensions match between actual and expected shapes.
+
+        Args:
+            actual_shape: The actual shape of the data
+            expected_shape: The expected shape
+            name: Name of the data for error messages
+
+        Raises:
+            ValueError: If dimensions don't match
+        """
+        # Check dimensions match
+        if len(actual_shape) != len(expected_shape):
+            raise ValueError(
+                f"{name} shape dimension mismatch: "
+                f"expected {len(expected_shape)} dimensions, got {len(actual_shape)}",
+            )
+
+        # Check each dimension matches
+        zipped_dims = zip(actual_shape, expected_shape, strict=False)
+        for i, (actual_dim, expected_dim) in enumerate(zipped_dims):
+            if expected_dim is not None and actual_dim != expected_dim:
+                raise ValueError(
+                    f"{name} shape mismatch at dimension {i}: "
+                    f"expected {expected_dim}, got {actual_dim}",
+                )
+
+    def validate_sample(self, config: dict | None = None) -> dict:
+        """Validates that the sample has the expected structure and data shapes.
+
+        Args:
+            config: Optional configuration dict with expected shapes and required fields.
+                If None, uses default validation rules.
+
+        Returns:
+            dict: Validation results with status and any validation errors.
+        """
+        # Default config if none
+        if config is None:
+            config = {
+                "required_keys": [
+                    GSPSampleKey.gsp,
+                    NWPSampleKey.nwp,
+                    SatelliteSampleKey.satellite_actual,
+                ],
+            }
+
+        # Check for required keys
+        for key in config.get("required_keys", []):
+            if key not in self._data:
+                raise ValueError(f"Missing required key: {key}")
+
+        expected_shapes = {}
+        if config.get("expected_shapes"):
+            expected_shapes.update(config["expected_shapes"])
```

Comment:
for me config is a COnfiguration, so we should add extra things like expected shape into it, or get it from there